File: src/ByteBite.Infrastructure/Persistence/Entities/Category.cs
using System;
using System.Collections.Generic;

namespace ByteBite.Infrastructure.Persistence.Entities;

/// <summary>
/// 商品分类表
/// </summary>
public partial class Category
{
    public Guid Id { get; set; }

    public Guid StoreId { get; set; }

    public string Name { get; set; } = null!;

    public string? Icon { get; set; }

    /// <summary>
    /// 分类类型：normal-普通, hot-热销, welfare-进店福利, combo-套餐
    /// </summary>
    public string CategoryType { get; set; } = null!;

    public int SortOrder { get; set; }

    public bool IsVisible { get; set; }

    /// <summary>
    /// 热销分类自动聚合的Top数量（仅hot类型有效）
    /// </summary>
    public int? HotTopCount { get; set; }

    public DateTime CreatedAt { get; set; }

    public DateTime UpdatedAt { get; set; }

    public DateTime? DeletedAt { get; set; }

    public virtual ICollection<ProductSalesStat> ProductSalesStats { get; set; } = new List<ProductSalesStat>();

    public virtual ICollection<Product> Products { get; set; } = new List<Product>();

    public virtual Store Store { get; set; } = null!;
}

File: web/src/types/models/category.ts
/** 分类信息DTO */
export interface CategoryDto {
  /** 分类ID */
  id: string
  /** 所属店铺ID */
  storeId: string
  /** 分类名称 */
  name: string
  /** 图标 */
  icon?: string
  /** 分类类型：normal-普通, hot-热销, welfare-进店福利, combo-套餐 */
  categoryType: string
  /** 排序序号 */
  sortOrder: number
  /** 是否可见 */
  isVisible: boolean
  /** 热销分类自动聚合的Top数量（仅hot类型有效） */
  hotTopCount?: number
  /** 分类下的商品数量 */
  productCount: number
}

/** 创建分类请求 */
export interface CreateCategoryRequest {
  /** 分类名称 */
  name: string
  /** 图标 */
  icon?: string
  /** 分类类型：normal-普通, hot-热销, welfare-进店福利, combo-套餐 */
  categoryType?: string
  /** 排序序号 */
  sortOrder?: number
  /** 热销分类自动聚合的Top数量 */
  hotTopCount?: number
}

/** 更新分类请求 */
export interface UpdateCategoryRequest {
  /** 分类名称 */
  name?: string
  /** 图标 */
  icon?: string
  /** 分类类型 */
  categoryType?: string
  /** 排序序号 */
  sortOrder?: number
  /** 是否可见 */
  isVisible?: boolean
  /** 热销分类自动聚合的Top数量 */
  hotTopCount?: number
}
